connectors/sources/mysql.py

```diff
@@ -21,12 +27,92 @@
     "TABLE_LAST_UPDATE_TIME": "SELECT UPDATE_TIME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = '{database}' AND TABLE_NAME = '{table}'",
 }
 DEFAULT_FETCH_SIZE = 50
-DEFAULT_RETRY_COUNT = 3
-DEFAULT_WAIT_MULTIPLIER = 2
+RETRIES = 3
+RETRY_INTERVAL = 2
 DEFAULT_SSL_DISABLED = True
 DEFAULT_SSL_CA = None
 
 
+def format_list(databases):
+    return ", ".join(databases)
+
+
+class MySQLAdvancedRulesValidator(AdvancedRulesValidator):
+    def __init__(self, source):
+        self.source = source
+
+    async def validate(self, advanced_rules):
+        databases = set(self.source.configured_databases())
+        databases_to_filter = set(advanced_rules.keys())
+
+        missing_databases = list(databases_to_filter - databases)
+
+        if len(missing_databases) > 0:
+            return SyncRuleValidationResult(
+                SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
+                validation_message=f"Non-configured databases: {format_list(missing_databases)}. Configured databases: {'None' if not len(databases) else format_list(databases)}.",
+            )
+
+        return await self._remote_validation(advanced_rules, databases_to_filter)
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+    )
+    async def _remote_validation(self, advanced_rules, databases_to_filter):
+        await self.source.ping()
```

Comment:
Do we want to ping again here? it's pinged by the protocol before we sync, I am just wondering if that's not going to be an extra ping on every attempt.

maybe `ping` can cache its result so we don't ping several times? and also invalidate that cache after some time

Reply:
I think we can just remove the `ping` call as we always ping before the validations. We could introduce ping caching with another PR, but I think this would be too much scope for this PR.

Reply:
I remember why I introduced `ping`. Ping actually instantiates the connection pool, if it's not instantiated. As validating can also happen outside of the context of a sync I've used to ping to make sure we can actually talk to the data source. I've noted it down and will create a tech-debt issue to separate the connection pool instantiation from `ping`.

So I would keep `ping` for now and tackle it in between documentation/test plan creation.

Reply:
yeah ok. Even if we have a lazy instantiation of pools, this should be an internal implementation detail, and  `ping` should not be mandatory to create the pool -- even if it's a mandatory step for the SyncService.

Reply:
Tracked by: https://github.com/elastic/enterprise-search-team/issues/3821